addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -1538,3 +1538,32 @@ def test_inventory_admin_no_backorder_not_own_sale_order(self):
         wizard = Form(self.env[(res_dict.get('res_model'))].with_user(inventory_admin_user).with_context(
             res_dict['context'])).save()
         wizard.with_user(inventory_admin_user).process_cancel_backorder()
+
+    def test_reduce_qty_ordered_no_backorder(self):
+        """
+        When validating a reduced picking, declining a backorder then reducing the quantity ordered on the SO line
+        to match the quantity delivered, make sure that no additional picking is created.
+        Steps:
+        - create a SO for 3 units of product A
+        - validate and open the delivery picking
+        - put 2 in quantity done, validate
+        - do NOT create a backorder
+        - return on the SO, put 2 in ordered quantity field
+        - save the SO
```

Comment:
no need to write exactly what you're doing in the test (since it can already be seen in the test + if the test changes then this description might not make sense anymore)